bindings/tflite/java/org/tensorflow/lite/Interpreter.java

```diff
@@ -0,0 +1,436 @@
+  }
+
+  /** Gets the number of input tensors. */
+  public int getInputTensorCount() {
+    return nativeInputTensorCount();
+  }
+
+  /**
+   * Gets index of an input given the op name of the input.
+   *
+   * @throws IllegalArgumentException if {@code opName} does not match any input in the model used
+   *     to initialize the {@link Interpreter}.
+   */
+  public int getInputIndex(String opName) {
+    for (int i = 0; i < getInputTensorCount(); ++i) {
+      if (getInputTensor(i).name().equals(opName)) {
+        return i;
+      }
+    }
+    return -1;
+  }
+
+  /**
+   * Gets the Tensor associated with the provdied input index.
+   *
+   * @throws IllegalArgumentException if {@code inputIndex} is negtive or is not smaller than the
+   *     number of model inputs.
+   */
+  public Tensor getInputTensor(int index) {
+    if (index < 0 || index >= inputTensors.length) {
+      throw new IllegalArgumentException(String.format("Invalid input Tensor index: %d", index));
+    }
+    Tensor inputTensor = inputTensors[index];
+    if (inputTensor == null) {
+      inputTensor = inputTensors[index] = Tensor.inputFromIndex(nativeAddress, index);
+    }
+    return inputTensor;
+  }
+
+  /** Gets the number of output Tensors. */
+  public int getOutputTensorCount() {
+    return nativeOutputTensorCount();
+  }
+
+  /** Gets index of an output given the op name of the output or -1 if not found. */
+  public int getOutputIndex(String opName) {
+    for (int i = 0; i < getOutputTensorCount(); ++i) {
+      if (getOutputTensor(i).name().equals(opName)) {
+        return i;
+      }
+    }
+    return -1;
+  }
+
+  /**
+   * Gets the Tensor associated with the provdied output index.
+   *
+   * <p>Note: Output tensor details (e.g., shape) may not be fully populated until after inference
+   * is executed. If you need updated details *before* running inference (e.g., after resizing an
+   * input tensor, which may invalidate output tensor shapes), use {@link #allocateTensors()} to
+   * explicitly trigger allocation and shape propagation. Note that, for graphs with output shapes
+   * that are dependent on input *values*, the output shape may not be fully determined until
+   * running inference.
+   *
+   * @throws IllegalArgumentException if {@code outputIndex} is negtive or is not smaller than the
+   *     number of model outputs.
+   */
+  public Tensor getOutputTensor(int index) {
+    if (index < 0 || index >= outputTensors.length) {
+      throw new IllegalArgumentException(String.format("Invalid output Tensor index: %d", index));
+    }
+    Tensor outputTensor = outputTensors[index];
+    if (outputTensor == null) {
+      outputTensor = outputTensors[index] = Tensor.outputFromIndex(nativeAddress, index);
+    }
+    return outputTensor;
+  }
+
+  /** Returns native inference timing, or -1 if inference isn't complete yet. */
+  public Long getLastNativeInferenceDurationNanoseconds() {
+    return inferenceDurationNanoseconds;
+  }
```

Comment:
Ew.. shame https://www.tensorflow.org/lite/api_docs/java/org/tensorflow/lite/Interpreter has this also. This is a very primitive metric to be exposing right alongside the other public methods. User applications can get this measurement themselves too... unless TFLite is adding substantial, non-JNI, overhead to their `run` -> `nativeInvoke` path.

Reply:
Yeah, its from their `Interpreter.java`. It wasn't hard to replicate and I didn't see a reason to omit it.